### PR TITLE
Fixes to MIDI playback

### LINKED ISSUE #312: UI improvements of the Play Audio button
- [x]  There should be a `stop` function (button).
- [x] The `Play Audio` button should be deactivated until the chant finishes or it is stopped.
- [x] The `Play Audio` button should be absent when there's no Volpiano.
- [x] Fix playback issues in Safari
- [x] playback should handle liquescents (right now has a gap)
- [x] playback should hand b-flats (currently treats them like naturals)
- [x] could stand to be a hair faster...(takes a long time to play the chants, esp. since you can't stop/pause)
- [x] maybe treat | as a rest?
- [x] Notes play on top of each other when playing chant for first time after page load
- [x] Play button does not have unique id (plays first expanded chant on page)

### PR DESCRIPTION
**Pull request by dchiller**
Closes #312. 

Makes the following additions/fixes:
- Introduces a stop button to halt midi playback.
- Adjusts behaviour of the play button to prevent multiple simultaneous/overlapping plays.
- Speeds up chant playback.
- Modifies plug-in load process to reduce speed of load time on playback initiation and improve page load consistency.
- Introduces handling of liquescent neumes, flats, and barlines.
- Adds support for browsers using mpeg.
- Removes Play button when no volpiano is available.
- Fixes playback timing to remove note bunching at the beginning of playback.

### CODE REVIEW

File: nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js [hunk 1]
```diff
@@ -53,6 +115,49 @@ function CustomSplit(str, delimiter, removeEmptyItems) {
...
+
+// A basic function to handle flats.
+// Flats are applied to B's that immediately follow a flat
+// along with any B's in the same text syllable, unless
+// a natural cancels it.
+function handleFlats (inputStr){
+	var flatReplDict = {};
+	flatReplDict["y"] = ["b","t"];
+	flatReplDict["i"] = ["j","u"];
+	flatReplDict["z"] = ["q","v"];
```
Comment (jacobdgm) [inline nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js:138]: can you add comments to each of these lines, something like `B4 -> Bb4`?

File: nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js [hunk 2]
```diff
@@ -4,24 +4,76 @@ import { parseVolpianoSyllables } from 'utils/VolpianoDisplayHelper';
...
-
-function dynamicallyLoadScript(url) {
+function dynamicallyLoadScript(url, onLoadFunction) {
 	var script = document.createElement("script");  // create a script DOM node
 	script.src = url;  // set its src to the provided URL
-	document.head.appendChild(script);  // add it to the end of the head section of the page (could change 'head' to 'body' to add it to the end of the body section instead)
+	if (onLoadFunction){
+		script.onload = onLoadFunction;
+	}
+	document.body.appendChild(script); // add it to the end of the head section of the page (could change 'head' to 'body' to add it to the end of the body section instead)
```
Comment (jacobdgm) [inline nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js:13]: comment needs to be updated

Comment (dchiller) [reply nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js:13]: good catch

File: nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js (hunk 1 above)
Comment (dchiller) [reply nginx/public/node/frontend/public/js/app/manuscript-detail/folio/ChantRecordView.js:138]: yes

**Review by jacobdgm — APPROVED** [review]
looks fine to me

**Pull request merged**